iset.mm

```diff
@@ -60958,6 +61008,23 @@ the cancellation property (fifth hypothesis), show that the relation
       UAUBUCUOURVDIVCUDULUQBUNUENUFUGUQAUMUHUIUJRUK $.
   $}
 
+  ${
+    $d A m n $.  $d B m n $.  $d V n $.
+    $( Adding a singleton to a finite set yields a finite set.  (Contributed by
+       Jim Kingdon, 3-Feb-2022.) $)
+    unsnfi $p |- ( ( A e. Fin /\ B e. V /\ -. B e. A )
+        -> ( A u. { B } ) e. Fin ) $=
+      ( vn vm cfn wcel wn cv cen wbr csn cun com wrex isfi wa ad2antrl cin c0
```

Comment:
Classically, you do not need B e. V nor -. B e. A (which you could join as B e. ( V \ A )). Do you need both intuitionistically ?
Maybe add also to iset: $p |- ( [what is needed] -> { A } e. Fin )

Reply:
> Classically, you do not need B e. V nor -. B e. A . . .
>  Do you need both intuitionistically ? 

Your intuition is perhaps mapping more to "subfinite" than "finite" (and those are equivalent given excluded middle) - there's a lot more on this in section 8.1 of [AczelRathjen] or http://nlab-pages.s3.us-east-2.amazonaws.com/nlab/show/finite%20set for example.

In this example, to say `( A u. { B } ) e. Fin` means you are mapping `( A u. { B } )` to a natural number, and whether that number is the size of ` A ` or the size of ` A ` plus one will tell which case we are dealing with. I think that could be said as `( A u. { B } ) e. Fin -> DECID ( B e. _V /\ B e. A )` or if not something along those lines.

There's at least a bit of a meta-point too - the current proof does use both (`-. B e. A` to apply `unen` and `B e. V` to apply `en2sn`). It is interesting to ponder whether a result can be strengthened but we will sometimes prove a result which is not as strong as what might be possible later (although this is sort of a point for another day if the reasoning above, about why the `B e. _V ` and ` B e. A` conditions are needed, holds up).

> (which you could join as B e. ( V \ A )).

That is what I originally wrote but I was thinking that the point of writing `B e. V` rather than `B e. _V` is that the former is often already present (in the form of `B e. _om` or whatever) and that having to break apart the `( V \ A )` may just end up an inconvenience whenever this is applied. We can always change it later, or have both forms, if it seems like `B e. ( V \ A )` is often a natural way to express this.

> Maybe add also to iset: $p |- ( [what is needed] -> { A } e. Fin )

If I understand the question this is http://us.metamath.org/ileuni/snfig.html . I mean we could also prove ` -. A e. _V -> { A } e. Fin` but that's pretty much just a restatement of http://us.metamath.org/ileuni/0fin.html and http://us.metamath.org/ileuni/snprc.html and it isn't like we could use http://us.metamath.org/mpeuni/pm2.61.html to combine the `A e. _V` and `-. A e. _V` cases so I'm not sure there's any particular reason to add the proper class version too.

Reply:
> `( A u. { B } ) e. Fin -> DECID ( B e. _V /\ B e. A )` or if not something along those lines.

That would be a nice addition (you probably meant -. B e. A). I do not know it it's true. An easier goal might be ` B e. V -> ( ( A u. { B } ) e. Fin -> DECID B e. A )`

Reply:
I don't think this is true. Counterexample: let `X = { x e. 1o | p }` as usual and then let `A = X` and `B = (/)`. Then `(A u. {B}) = 1o` which is finite but `B e. A` is equivalent to `p`.

Reply:
> I don't think this is true. Counterexample: let `X = { x e. 1o | p }` as usual and then let `A = X` and `B = (/)`. Then `(A u. {B}) = 1o` which is finite but `B e. A` is equivalent to `p`.
> 
> I believe it is true if you assume `A` is also finite, because as Jim said you can case on whether `|A| = |(A u. {B})|` or `|A| + 1 = |(A u. {B})|`.

I'll make a follow up pull request for this, but you are right, we weren't being very explicit. I did indeed intend the case where `A e. Fin`.

Reply:
Follow up pull request is https://github.com/metamath/set.mm/pull/2477